src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/vm/AzureVMsProvider.java

```diff
@@ -0,0 +1,295 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.connector.iaas.cloud.provider.azure.vm;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.apache.log4j.Logger;
+import org.ow2.proactive.connector.iaas.cloud.TagManager;
+import org.ow2.proactive.connector.iaas.cloud.provider.CloudProvider;
+import org.ow2.proactive.connector.iaas.cloud.provider.azure.AzureProvider;
+import org.ow2.proactive.connector.iaas.cloud.provider.azure.AzureProviderNetworkingUtils;
+import org.ow2.proactive.connector.iaas.cloud.provider.azure.AzureProviderUtils;
+import org.ow2.proactive.connector.iaas.cloud.provider.azure.AzureServiceCache;
+import org.ow2.proactive.connector.iaas.model.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.google.common.collect.Lists;
+import com.microsoft.azure.management.Azure;
+import com.microsoft.azure.management.compute.*;
+import com.microsoft.azure.management.network.*;
+import com.microsoft.azure.management.network.Network;
+import com.microsoft.azure.management.network.model.HasPrivateIpAddress;
+import com.microsoft.azure.management.network.model.HasPublicIpAddress;
+import com.microsoft.azure.management.resources.ResourceGroup;
+import com.microsoft.azure.management.resources.fluentcore.arm.Region;
+import com.microsoft.azure.management.resources.fluentcore.model.Creatable;
+import com.microsoft.azure.management.resources.fluentcore.utils.SdkContext;
+
+import lombok.Getter;
+
+
+/**
+ * Provides Microsoft Azure clouds' management using the official java SDK.
+ *
+ * This class has been tested by ActiveEon to be thread safe (using Azure SDK release version 1.0.0-beta5).
+ * However this need to be carefully double-checked after every SDK upgrades, as mentioned by Microsoft:
+ * ------------------------------------------------------------------------------------------------------------
+ * We do not make any thread-safety guarantees about our libraries. We also do not test them for thread-safety.
+ * Methods that are currently thread-safe may be thread-unsafe in future versions.
+ * ------------------------------------------------------------------------------------------------------------
+ *
+ * @author ActiveEon Team
+ * @since 01/03/17
+ */
+@Component
+public class AzureVMsProvider extends AzureProvider {
+
+    @Getter
+    protected final String type = "azure";
+
+    private final Logger logger = Logger.getLogger(AzureVMsProvider.class);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Make the "logger" logger private static final. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1312)